src/main/java/life/qbic/qpostman/common/functions/FileFilter.java

```diff
@@ -1,31 +1,55 @@
 package life.qbic.qpostman.common.functions;
 
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
 import java.util.ArrayList;
 import java.util.List;
 import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
 import life.qbic.qpostman.common.structures.DataFile;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 /**
  * A class implementing the {@link Predicate} interface and providing file filtering functionality based on suffixes.
  */
 public class FileFilter implements Predicate<DataFile> {
+    private static final Logger log = LogManager.getLogger(FileFilter.class);
 
     private final List<String> suffixes;
     private final boolean caseSensitive;
+    private final Pattern pattern;
 
     public static FileFilter create() {
-        return new FileFilter(new ArrayList<>(), false);
+        return new FileFilter(new ArrayList<>(), false, null);
     }
 
     public FileFilter withSuffixes(List<String> suffixes) {
         var temp = new ArrayList<>(this.suffixes);
         temp.addAll(suffixes);
-        return new FileFilter(temp, caseSensitive);
+        return new FileFilter(temp, caseSensitive, pattern);
+    }
+
+    public FileFilter withPattern(String pattern) {
+        if (isNull(pattern)) {
+            return new FileFilter(suffixes, caseSensitive, null);
+        }
+        Pattern compiledPattern = null;
+        try {
+            compiledPattern = Pattern.compile(pattern);
+        } catch (PatternSyntaxException e) {
+            log.error("The pattern %s is malformed. %s".formatted(e.getPattern(), e.getMessage()));
+            System.exit(1);
```

Comment:
I think you should not exit the application this deep in an application. I suggest you either rethrow the exception or throw a more general exception and handle the exit code in the main class of the application.

Reply:
fixed in #178